flow/__main__.py

```diff
@@ -46,6 +50,46 @@ def main_init(args):
         )
 
 
+def main_template_create(args):
+    """Create custom template based on a specified or detected environment."""
+    if args.extends is None:
+        extend_template = environment.get_environment().template
+    else:
+        extend_template = args.extends[0]
+
+    project = get_project()
+    template_directory = project.fn(project._config.get("template_dir", "templates"))
+    os.makedirs(template_directory, exist_ok=True)
+
+    # grab and render custom template
+    jinja_env = jinja2.Environment(loader=jinja2.PackageLoader("flow"))
+    custom_template = jinja_env.get_template("custom_script_template.sh")
+    new_template = custom_template.render(extend_template=extend_template).strip(
+        string.whitespace
+    )
+
+    # If name is the default then it is a string if not it gets passed as a list
+    script_name = args.name if isinstance(args.name, str) else args.name[0]
+    script_path = os.path.join(template_directory, script_name)
+    try:
+        with open(script_path, "x") as fh:
+            fh.write(new_template)
+    except OSError as error:
+        if error.errno == errno.EEXIST:
+            logger.error(
+                f"Error while trying to create custom template. Delete '{script_path}' "
+                f"first and rerun command."
+            )
+        else:
+            logger.error(f"Error while trying to create custom template: '{error}'.")
```

Comment:
How could we trigger this error in our tests, so we have 100% coverage? 😉 Don't spend a lot of time on this, unless you have an idea of how to trigger the error. Perhaps this would be triggered if the user lacks permissions to create a new file?

Reply:
That would work, but I don't know of an easy way to trigger this except to potentially overwrite the `open` function to just raise an `OSError`.